### PR TITLE
Cache / Permissions: Prevent server crash when command times out

### PR DESCRIPTION
**Pull request by joselcvarela**
## Description

When getting values from cache takes more than `CACHE_${STORE}_COMMAND_TIMEOUT` an error is thrown.
While this is properly handled for `items` [here](https://github.com/directus/directus/blob/main/api/src/middleware/cache.ts#L25-L31) it is not being handled for permissions, which causes server to crash with promise rejection.
This PR adds a `try catch` block around the code which fetches the permissions from cache, so it returns nothing if fails but error is logged on console.

Closes ENG-351

## Type of Change

- [x] Bugfix
- [ ] Improvement
- [ ] New Feature
- [ ] Refactor / codestyle updates
- [ ] Other, please describe:

## Requirements Checklist

- [ ] New / updated tests are included
- [x] All tests are passing locally
- [x] Performed a self-review of the submitted code

If adding a new feature:

- [ ] Documentation was added/updated. PR:

### CODE REVIEW

File: api/src/utils/get-permissions.ts [hunk 1]
```diff
@@ -21,7 +22,13 @@ export async function getPermissions(accountability: Accountability, schema: Sch
 	const cacheKey = `permissions-${hash({ user, role, app, admin, share_scope })}`;
 
 	if (env.CACHE_PERMISSIONS !== false) {
-		const cachedPermissions = await getSystemCache(cacheKey);
+		let cachedPermissions = null;
```
Comment (azrikahar) [inline api/src/utils/get-permissions.ts:25]:
```suggestion
		let cachedPermissions;
```

Thoughts on _not_ setting it as null, similar to the code you have mentioned in your PR? 

https://github.com/directus/directus/blob/0122c4da4b2cdabd0fed830f27fbec0caa24071d/api/src/middleware/cache.ts#L23-L31

Of course I don't think there's any issue with `= null` 👍 just slightly gravitating towards not setting null personally, so wondering what are the general consensus around this 😄

**Review by azrikahar — COMMENTED** [review]
This is a tangent (and feel free to shift this to a separate discussion internally!), but I do notice here in get-permissions, we are still using `cache.get` and `cache.set` directly: 

https://github.com/directus/directus/blob/7c6f54f5dbfafa3888abded26c57907ac36bef59/api/src/utils/get-permissions.ts#L38-L40

https://github.com/directus/directus/blob/7c6f54f5dbfafa3888abded26c57907ac36bef59/api/src/utils/get-permissions.ts#L58

but `getCacheValue` and `setCacheValue` does do compressions:

https://github.com/directus/directus/blob/0122c4da4b2cdabd0fed830f27fbec0caa24071d/api/src/cache.ts#L64-L79

should we also use `getCacheValue`/`setCacheValue`, or could there be specific reasons where permissions cache couldn't/shouldn't be compressed?

File: api/src/utils/get-permissions.ts (hunk 1 above)
Comment (rijkvanzanten) [reply api/src/utils/get-permissions.ts]: Good catch :) Lets keep it consistent!

**Comment by rijkvanzanten** [comment]
Nice find @azrikahar! I'm pretty sure that's an oversight from a global find on `cache.get` instead where `cache?.get` was missed 🙂

**Review by rijkvanzanten — APPROVED** [review]

**Pull request merged**